enterprise/com/src/main/java/org/neo4j/com/storecopy/ToNetworkStoreWriter.java

```diff
@@ -45,20 +45,21 @@ public ToNetworkStoreWriter( ChannelBuffer targetBuffer, Monitors monitors )
 
     @Override
     public long write( String path, ReadableByteChannel data, ByteBuffer temporaryBuffer,
-            boolean hasData ) throws IOException
+            boolean hasData, int recordSize ) throws IOException
     {
         char[] chars = path.toCharArray();
         targetBuffer.writeShort( chars.length );
         Protocol.writeChars( targetBuffer, chars );
         targetBuffer.writeByte( hasData ? 1 : 0 );
         // TODO Make use of temporaryBuffer?
         BlockLogBuffer buffer = new BlockLogBuffer( targetBuffer, bufferMonitor );
-        long totalWritten = 2 + chars.length*2 + 1;
+        long totalWritten = Short.BYTES + chars.length* Character.BYTES + Byte.BYTES;
         if ( hasData )
         {
+            targetBuffer.writeInt( recordSize );
```

Comment:
It looks like you're changing the network protocol. For compatibility, and detection of incompatibility you have to increment protocol version. I think this is a case where the internal protocol version should be incremented, see https://github.com/neo4j/neo4j/blob/3.1/enterprise/com/src/main/java/org/neo4j/com/ProtocolVersion.java#L24